platforms/jvm/jacoco/src/main/java/org/gradle/internal/jacoco/AntJacocoCheck.java

```diff
@@ -46,70 +46,105 @@ public boolean apply(JacocoViolationRule rule) {
-                                    limitArgs.put("value", limit.getValue());
+                                for (final JacocoViolationRule rule : rules) {
+                                    Map<String, Object> ruleArgs = ImmutableMap.<String, Object>of("element", rule.getElement(), "includes", Joiner.on(':').join(rule.getIncludes()), "excludes", Joiner.on(':').join(rule.getExcludes()));
+                                    antBuilder.invokeMethod("rule", new Object[] {ruleArgs, new Closure<Object>(this, this) {
+                                        @SuppressWarnings("UnusedDeclaration")
+                                        public Object doCall(Object ignore) {
+                                            for (JacocoLimit limit : rule.getLimits()) {
+                                                Map<String, Object> limitArgs = new HashMap<String, Object>();
+                                                limitArgs.put("counter", limit.getCounter());
+                                                limitArgs.put("value", limit.getValue());
 
-                                    if (limit.getMinimum() != null) {
-                                        limitArgs.put("minimum", limit.getMinimum());
-                                    }
-                                    if (limit.getMaximum() != null) {
-                                        limitArgs.put("maximum", limit.getMaximum());
-                                    }
+                                                if (limit.getMinimum() != null) {
+                                                    limitArgs.put("minimum", limit.getMinimum());
+                                                }
+                                                if (limit.getMaximum() != null) {
+                                                    limitArgs.put("maximum", limit.getMaximum());
+                                                }
 
-                                    antBuilder.invokeMethod("limit", new Object[] {ImmutableMap.copyOf(limitArgs) });
+                                                antBuilder.invokeMethod("limit", new Object[] {ImmutableMap.copyOf(limitArgs) });
+                                            }
+                                            return Void.class;
+                                        }
+                                    }});
                                 }
-                                return null;
+                                return Void.class;
                             }
                         }});
                     }
-                    return null;
+
+                    return Void.class;
                 }
             }});
+        } catch (Exception e) {
+            String violations = getViolations(antBuilder);
+            throw new GradleException(violations == null ? e.getMessage() : violations);
         }
+
+        GFileUtils.touch(params.getDummyOutputFile().get().getAsFile());
```

Comment:
❓ What is this dummy output file?  Was this here in some other form before?

Reply:
It was here already, but done in the task: 
https://github.com/gradle/gradle/pull/32257/files#diff-4e419038d825328a402bfe565552ffa9a4242ef74afdf513fb356aa995789771L95